bear fails compilation in connection with apple's llvm 10

NOTE: I am using pre-released software, hence I do not "expect" anything but hope for the best. In this case, my hope is not matched by reality :)

### Expected Behavior
Running `make` without bear yields a proper binary - no problems building.

```
$ make
g++ -c -o test.o test.cpp
g++ -o test test.o
````

### Actual Behavior
Running `bear make` causes standard headers to be not found -- VERY similar to a failure of running `xcode-select --install`. However I did run properly multiple times - compilation only fails in connection with `bear`, it seems.  

```
$ bear make
g++ -c -o test.o test.cpp
In file included from test.cpp:1:
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdio.h:108:15: fatal error: 'stdio.h' file not found
#include_next <stdio.h>
              ^~~~~~~~~
1 error generated.
make: *** [test.o] Error 1
```

### Environment

- OS name & version:
macOS 10.14

- Bear version:
bear 2.3.11

- Any reference to the project that you running against:
Makefile:
```
test: test.o
	g++ -o test test.o

test.o: test.cpp
	g++ -c -o test.o test.cpp
```

test.cpp:
```
#include <stdio.h>

int main (int argc, char **argv)
{
	return 0;
}
```


- Any information about the compiler that this project is using:
```
$ g++ --version
Configured with: --prefix=/Applications/Xcode-beta.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.10.25.5)
Target: x86_64-apple-darwin18.0.0
Thread model: posix
InstalledDir: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

`bear -vvvv make` output:
```
bear: DEBUG: parse_args_for_intercept_build: Raw arguments ['/usr/local/bin/bear', '-vvvv', 'make']
bear: DEBUG: parse_args_for_intercept_build: Parsed arguments: Namespace(append=False, build=['make'], cc='ccache gcc -fcolor-diagnostics', cdb='compile_commands.json', cxx='ccache g++ -fcolor-diagnostics', libear='/usr/local/Cellar/bear/2.3.11/lib/bear/libear.dylib', verbose=4)
bear: DEBUG: run_build: run build ['make'], in environment: {'_fzf_orig_completion_ld': 'complete -o default -F %s ld #_longopt', 'LESS': '--RAW-CONTROL-CHARS', '_fzf_orig_completion_ln': 'complete -o default -F %s ln #_longopt', '_fzf_orig_completion_rsync': 'complete -o nospace -F %s rsync #_rsync', 'RTAGS_START': 'brew services start rtags', '_fzf_orig_completion_emacs': 'complete -F %s emacs #_filedir_xspec', '_fzf_orig_completion_ls': 'complete -o default -F %s ls #_longopt', 'VIRTUALENVWRAPPER_SCRIPT': '/usr/local/bin/virtualenvwrapper.sh', '_fzf_orig_completion_diff': 'complete -o default -F %s diff #_longopt', 'SHELL': '/bin/bash', 'COLORFGBG': '15;0', '_fzf_orig_completion_tail': 'complete -o default -F %s tail #_longopt', '_fzf_orig_completion_sed': 'complete -o default -F %s sed #_longopt', 'XPC_FLAGS': '0x0', 'ITERM_ORIG_PS1': '\\[\\e[1;34m\\]\\w\\[\\e[m\\]$(\n    # Figure out what repo we are in\n    _git_check\n\n    # Build the prompt!\n    prompt=""\n\n    # If we are in git ...\n    if [ -n "$_git_dir" ]; then\n        # find current branch\n        gitBranch=$(git symbolic-ref HEAD 2> /dev/null)\n        gitStatus=`git status`\n\n        # Figure out if we are rebasing\n        if [[ -d "$_git_dir/.git/rebase-apply" || -d "$_git_dir/.git/rebase-merge" ]]; then\n            is_rebase=1\n        fi\n\n        # Figure out current branch, or if we are bisecting, or lost in space\n        bisecting=""\n        if [ -z "$gitBranch" ]; then\n            if [ -n "$is_rebase" ]; then\n                rebase_prompt=" \\[$COLOR_LIGHT_CYAN\\]"\n                rebase_prompt=$rebase_prompt"\\[\x1b7\\]  \\[\x1b8\\]\\[$REBASE_TEXT\\] "\n                rebase_prompt=$rebase_prompt"\\[$COLOR_YELLOW\\]"\n            else\n                gitBranch="\\[$COLOR_RED\\]$NOBRANCH_TEXT\\[$COLOR_YELLOW\\]"\n            fi\n        fi\n\n        gitBranch=${gitBranch#refs/heads/}\n        #: git-svn\n        if [ -z "$bisect" ]; then\n            if [ -n "$_git_svn_dir" ]; then\n                gitBranch="\\[$COLOR_DARK_BLUE\\]git-svn\\[$COLOR_YELLOW\\] $gitBranch"\n            fi\n        fi\n\n\n        gitNochange=""\n        if [ -z "$is_rebase" ]; then\n            # changed *tracked* files in local directory?\n\t    gitChange=$(echo $gitStatus | ack \'modified:|deleted:|new file:\')\n            if [ -n "$gitChange" ]; then\n\t\tgitChange="\\[$COLOR_YELLOW\\] \\[\x1b7\\]  \\[\x1b8\\]\\[$DELTA_CHAR\\] "\n            else\n\t\tgitChange="\\[$COLOR_GREEN\\] \\[\x1b7\\]  \\[\x1b8\\]\\[$OK_CHAR\\] "\n            fi\n        fi\n\n        # output the branch and changed character if present\n        prompt=$prompt"\\[$COLOR_YELLOW\\] ("\n\n        if is_submodule; then\n            prompt=$prompt"\\[$COLOR_MAGENTA\\]$SUBMODULE_TEXT\\[$COLOR_YELLOW\\]"\n        fi\n\n        prompt=$prompt$prefix$gitBranch$bisecting\n        prompt=$prompt"$gitChange\\[$COLOR_YELLOW\\])\\[$COLOR_RESET\\]"\n\n        # any conflicts? (sed madness is to remove line breaks)\n        files=$(git ls-files -u | cut -f 2 | sort -u | sed \'3,1000d\' | sed -e :a -e \'$!N;s/\\\\n/, /;ta\' -e \'P;D\')\n    fi\n\n\n    # Show conflicted files if any\n    if [ -n "$files" ]; then\n        prompt=$prompt" \\[$COLOR_RED\\](\\[$COLOR_YELLOW\\]"\n        prompt=$prompt"\\[\x1b7\\]  \\[\x1b8\\]\\[$COLOR_YELLOW\\]\\[$CONFLICT_CHAR\\] "\n        prompt=$prompt"\\[$COLOR_RED\\] ${files})"\n    fi\n\n    echo -e $prompt)\\[\x1b[m\\] \\[\\e[1;32m\\]$\\[\\e[m\\]\\[\\e[1;37m\\] ', 'JAVA_HOME': '/Library/Java/JavaVirtualMachines/jdk1.8.0_20.jdk/Contents/Home', 'VAGRANT_DEFAULT_PROVIDER': 'vmware_fusion', 'WORKON_HOME': '/Users/till/.virtualenvs', '_fzf_orig_completion_git': 'complete -o bashdefault -o default -o nospace -F %s git #__git_wrap__git_main', '_fzf_orig_completion_javac': 'complete -F %s javac #_javac', '_fzf_orig_completion_mv': 'complete -o default -F %s mv #_longopt', '_fzf_orig_completion_bunzip2': 'complete -F %s bunzip2 #_filedir_xspec', '_fzf_orig_completion_cat': 'complete -o default -F %s cat #_longopt', '_fzf_orig_completion_scp': 'complete -o nospace -F %s scp #_scp', 'HOMEBREW_GITHUB_API_TOKEN': '8e30efb3dbd041aa628919876456e1bc25d92fe1', '_fzf_orig_completion_ssh': 'complete -F %s ssh #_ssh', 'CXX': 'ccache g++ -fcolor-diagnostics', 'GOPATH': '/Users/till/Development/go-workspace', 'USER': 'till', 'PS1': '\\[\\e[1;34m\\]\\w\\[\\e[m\\]$(\n    # Figure out what repo we are in\n    _git_check\n\n    # Build the prompt!\n    prompt=""\n\n    # If we are in git ...\n    if [ -n "$_git_dir" ]; then\n        # find current branch\n        gitBranch=$(git symbolic-ref HEAD 2> /dev/null)\n        gitStatus=`git status`\n\n        # Figure out if we are rebasing\n        if [[ -d "$_git_dir/.git/rebase-apply" || -d "$_git_dir/.git/rebase-merge" ]]; then\n            is_rebase=1\n        fi\n\n        # Figure out current branch, or if we are bisecting, or lost in space\n        bisecting=""\n        if [ -z "$gitBranch" ]; then\n            if [ -n "$is_rebase" ]; then\n                rebase_prompt=" \\[$COLOR_LIGHT_CYAN\\]"\n                rebase_prompt=$rebase_prompt"\\[\x1b7\\]  \\[\x1b8\\]\\[$REBASE_TEXT\\] "\n                rebase_prompt=$rebase_prompt"\\[$COLOR_YELLOW\\]"\n            else\n                gitBranch="\\[$COLOR_RED\\]$NOBRANCH_TEXT\\[$COLOR_YELLOW\\]"\n            fi\n        fi\n\n        gitBranch=${gitBranch#refs/heads/}\n        #: git-svn\n        if [ -z "$bisect" ]; then\n            if [ -n "$_git_svn_dir" ]; then\n                gitBranch="\\[$COLOR_DARK_BLUE\\]git-svn\\[$COLOR_YELLOW\\] $gitBranch"\n            fi\n        fi\n\n\n        gitNochange=""\n        if [ -z "$is_rebase" ]; then\n            # changed *tracked* files in local directory?\n\t    gitChange=$(echo $gitStatus | ack \'modified:|deleted:|new file:\')\n            if [ -n "$gitChange" ]; then\n\t\tgitChange="\\[$COLOR_YELLOW\\] \\[\x1b7\\]  \\[\x1b8\\]\\[$DELTA_CHAR\\] "\n            else\n\t\tgitChange="\\[$COLOR_GREEN\\] \\[\x1b7\\]  \\[\x1b8\\]\\[$OK_CHAR\\] "\n            fi\n        fi\n\n        # output the branch and changed character if present\n        prompt=$prompt"\\[$COLOR_YELLOW\\] ("\n\n        if is_submodule; then\n            prompt=$prompt"\\[$COLOR_MAGENTA\\]$SUBMODULE_TEXT\\[$COLOR_YELLOW\\]"\n        fi\n\n        prompt=$prompt$prefix$gitBranch$bisecting\n        prompt=$prompt"$gitChange\\[$COLOR_YELLOW\\])\\[$COLOR_RESET\\]"\n\n        # any conflicts? (sed madness is to remove line breaks)\n        files=$(git ls-files -u | cut -f 2 | sort -u | sed \'3,1000d\' | sed -e :a -e \'$!N;s/\\\\n/, /;ta\' -e \'P;D\')\n    fi\n\n\n    # Show conflicted files if any\n    if [ -n "$files" ]; then\n        prompt=$prompt" \\[$COLOR_RED\\](\\[$COLOR_YELLOW\\]"\n        prompt=$prompt"\\[\x1b7\\]  \\[\x1b8\\]\\[$COLOR_YELLOW\\]\\[$CONFLICT_CHAR\\] "\n        prompt=$prompt"\\[$COLOR_RED\\] ${files})"\n    fi\n\n    echo -e $prompt)\\[\x1b[m\\] \\[\\e[1;32m\\]$\\[\\e[m\\]\\[\\e[1;37m\\] ', '_fzf_orig_completion_sort': 'complete -o default -F %s sort #_longopt', 'PYENV_VIRTUALENV_INIT': '1', '_fzf_orig_completion_gvim': 'complete -F %s gvim #_filedir_xspec', 'SHLVL': '1', '_fzf_orig_completion_unzip': 'complete -F %s unzip #_filedir_xspec', '_fzf_orig_completion_gzip': 'complete -F %s gzip #_gzip', '_fzf_orig_completion_tar': 'complete -F %s tar #_tar', 'EDITOR': 'subl -w', '_fzf_orig_completion_svn': "complete -o default -X '@(*/.svn|*/.svn/|.svn|.svn/)' -F %s svn #_svn", 'TERM_PROGRAM': 'iTerm.app', 'TERM_SESSION_ID': 'w0t0p0:02C68BD9-48F1-45C0-AC8F-F335B76DCC90', '_fzf_orig_completion_patch': 'complete -o default -F %s patch #_longopt', 'XPC_SERVICE_NAME': '0', '_fzf_orig_completion_rm': 'complete -o default -F %s rm #_longopt', 'ITERM_PROFILE': 'Default', 'GTEST_FILTER': '-IOSwitchboardServerTest*', '_fzf_orig_completion_ftp': 'complete -F %s ftp #_known_hosts', '_': '/usr/local/bin/bear', '_fzf_orig_completion_gcc': 'complete -F %s gcc #_gcc', '_fzf_orig_completion_java': 'complete -F %s java #_java', '_fzf_orig_completion_cd': 'complete -o nospace -F %s cd #_cd', 'COLORTERM': 'truecolor', '_fzf_orig_completion_cp': 'complete -o default -F %s cp #_longopt', '_fzf_orig_completion_chown': 'complete -F %s chown #_chown', '_fzf_orig_completion_python': 'complete -F %s python #_python', 'COMPILER_DEFAULTS': '-fcolor-diagnostics', 'TERM_PROGRAM_VERSION': '3.1.6beta5', 'DYLD_INSERT_LIBRARIES': '/usr/local/Cellar/bear/2.3.11/lib/bear/libear.dylib', 'INTERCEPT_BUILD_TARGET_DIR': '/var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w', '_fzf_orig_completion_wc': 'complete -o default -F %s wc #_longopt', 'HOME': '/Users/till', 'DISPLAY': '/private/tmp/com.apple.launchd.w1zmMs1L3K/org.macosforge.xquartz:0', '_fzf_orig_completion_rmdir': 'complete -o default -F %s rmdir #_longopt', 'LANG': 'en_US.UTF-8', 'Apple_PubSub_Socket_Render': '/private/tmp/com.apple.launchd.07eZ7XkLid/Render', '_fzf_orig_completion_jar': 'complete -F %s jar #_jar', 'TMPDIR': '/var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/', '_fzf_orig_completion_find': 'complete -F %s find #_find', 'CC': 'ccache gcc -fcolor-diagnostics', '_fzf_orig_completion_awk': 'complete -o default -F %s awk #_longopt', 'LSCOLORS': 'GxFxCxDxBxegedabagaced', '_fzf_orig_completion_sftp': 'complete -F %s sftp #_sftp', '_fzf_orig_completion_du': 'complete -o default -F %s du #_longopt', 'VIRTUALENVWRAPPER_HOOK_DIR': '/Users/till/.virtualenvs', '_fzf_orig_completion_g__': 'complete -F %s g++ #_gcc', '__CF_USER_TEXT_ENCODING': '0x1F5:0x0:0x0', '_fzf_orig_completion_perl': 'complete -F %s perl #_perl', 'PERL_MB_OPT': '--install_base "/Users/till/perl5"', '_fzf_orig_completion_uniq': 'complete -o default -F %s uniq #_longopt', '_fzf_orig_completion_export': 'complete -o default -o nospace -F %s export #_export', '_fzf_orig_completion_kill': 'complete -F %s kill #_kill', 'PERL_MM_OPT': 'INSTALL_BASE=/Users/till/perl5', 'LOGNAME': 'till', 'PATH': '/usr/local/Cellar/pyenv-virtualenv/1.1.1/shims:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Users/till/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/opt/X11/bin:/usr/local/opt/fzf/bin', '_fzf_orig_completion_tee': 'complete -o default -F %s tee #_longopt', '_fzf_orig_completion_view': 'complete -F %s view #_filedir_xspec', 'TERM': 'xterm-color', 'VIRTUALENVWRAPPER_WORKON_CD': '1', '_fzf_orig_completion_vim': 'complete -F %s vim #_filedir_xspec', 'PWD': '/Users/till/Desktop', '_fzf_orig_completion_curl': 'complete -o default -F %s curl #_longopt', 'ITERM_SESSION_ID': 'w0t0p0:02C68BD9-48F1-45C0-AC8F-F335B76DCC90', '_fzf_orig_completion_telnet': 'complete -F %s telnet #_known_hosts', 'DYLD_FORCE_FLAT_NAMESPACE': '1', '_fzf_orig_completion_head': 'complete -o default -F %s head #_longopt', '_fzf_orig_completion_less': 'complete -o default -F %s less #_longopt', 'SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.VFc6iARuQ8/Listeners', '_fzf_orig_completion_bzip2': 'complete -F %s bzip2 #_bzip2', '_fzf_orig_completion_vi': 'complete -F %s vi #_filedir_xspec', '_fzf_orig_completion_grep': 'complete -o default -F %s grep #_longopt', 'VIRTUALENVWRAPPER_PROJECT_FILENAME': '.project', 'CONFIGURE_DEFAULTS': '--disable-java --disable-python --enable-libevent --enable-ssl --disable-werror --with-ssl=/usr/local/opt/openssl', 'OLDPWD': '/Users/till', 'GPGKEY': 'ED50F391', '_fzf_orig_completion_gunzip': 'complete -F %s gunzip #_filedir_xspec', 'CLICOLOR': 'true', 'HISTCONTROL': '', 'RTAGS_STOP': 'brew services stop rtags', 'ITERM_PREV_PS1': '\\[\x1b]133;D;$?\x07\x1b]133;A\x07\\]\\[\\e[1;34m\\]\\w\\[\\e[m\\]$(\n    # Figure out what repo we are in\n    _git_check\n\n    # Build the prompt!\n    prompt=""\n\n    # If we are in git ...\n    if [ -n "$_git_dir" ]; then\n        # find current branch\n        gitBranch=$(git symbolic-ref HEAD 2> /dev/null)\n        gitStatus=`git status`\n\n        # Figure out if we are rebasing\n        if [[ -d "$_git_dir/.git/rebase-apply" || -d "$_git_dir/.git/rebase-merge" ]]; then\n            is_rebase=1\n        fi\n\n        # Figure out current branch, or if we are bisecting, or lost in space\n        bisecting=""\n        if [ -z "$gitBranch" ]; then\n            if [ -n "$is_rebase" ]; then\n                rebase_prompt=" \\[$COLOR_LIGHT_CYAN\\]"\n                rebase_prompt=$rebase_prompt"\\[\x1b7\\]  \\[\x1b8\\]\\[$REBASE_TEXT\\] "\n                rebase_prompt=$rebase_prompt"\\[$COLOR_YELLOW\\]"\n            else\n                gitBranch="\\[$COLOR_RED\\]$NOBRANCH_TEXT\\[$COLOR_YELLOW\\]"\n            fi\n        fi\n\n        gitBranch=${gitBranch#refs/heads/}\n        #: git-svn\n        if [ -z "$bisect" ]; then\n            if [ -n "$_git_svn_dir" ]; then\n                gitBranch="\\[$COLOR_DARK_BLUE\\]git-svn\\[$COLOR_YELLOW\\] $gitBranch"\n            fi\n        fi\n\n\n        gitNochange=""\n        if [ -z "$is_rebase" ]; then\n            # changed *tracked* files in local directory?\n\t    gitChange=$(echo $gitStatus | ack \'modified:|deleted:|new file:\')\n            if [ -n "$gitChange" ]; then\n\t\tgitChange="\\[$COLOR_YELLOW\\] \\[\x1b7\\]  \\[\x1b8\\]\\[$DELTA_CHAR\\] "\n            else\n\t\tgitChange="\\[$COLOR_GREEN\\] \\[\x1b7\\]  \\[\x1b8\\]\\[$OK_CHAR\\] "\n            fi\n        fi\n\n        # output the branch and changed character if present\n        prompt=$prompt"\\[$COLOR_YELLOW\\] ("\n\n        if is_submodule; then\n            prompt=$prompt"\\[$COLOR_MAGENTA\\]$SUBMODULE_TEXT\\[$COLOR_YELLOW\\]"\n        fi\n\n        prompt=$prompt$prefix$gitBranch$bisecting\n        prompt=$prompt"$gitChange\\[$COLOR_YELLOW\\])\\[$COLOR_RESET\\]"\n\n        # any conflicts? (sed madness is to remove line breaks)\n        files=$(git ls-files -u | cut -f 2 | sort -u | sed \'3,1000d\' | sed -e :a -e \'$!N;s/\\\\n/, /;ta\' -e \'P;D\')\n    fi\n\n\n    # Show conflicted files if any\n    if [ -n "$files" ]; then\n        prompt=$prompt" \\[$COLOR_RED\\](\\[$COLOR_YELLOW\\]"\n        prompt=$prompt"\\[\x1b7\\]  \\[\x1b8\\]\\[$COLOR_YELLOW\\]\\[$CONFLICT_CHAR\\] "\n        prompt=$prompt"\\[$COLOR_RED\\] ${files})"\n    fi\n\n    echo -e $prompt)\\[\x1b[m\\] \\[\\e[1;32m\\]$\\[\\e[m\\]\\[\\e[1;37m\\] \\[\x1b]133;B\x07\\]'}
g++ -c -o test.o test.cpp
In file included from test.cpp:1:
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdio.h:108:15: fatal error: 'stdio.h' file not found
#include_next <stdio.h>
              ^~~~~~~~~
1 error generated.
make: *** [test.o] Error 1
bear: DEBUG: run_build: build finished with exit code: 2
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.Tv8mDg
bear: DEBUG: _split_command: input was: [u'/Applications/Xcode-beta.app/Contents/Developer/usr/bin/g++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.ivdAJT
bear: DEBUG: _split_command: input was: [u'g++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.pgKSbx
bear: DEBUG: _split_command: input was: [u'g++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.xCo9P4
bear: DEBUG: _split_command: input was: [u'/usr/bin/xcrun', u'clang++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.qPIqbY
bear: DEBUG: _split_command: input was: [u'/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.ogdbmu
bear: DEBUG: _split_command: input was: [u'g++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.frHJVm
bear: DEBUG: _split_command: input was: [u'g++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.kAKraR
bear: DEBUG: _split_command: input was: [u'/Applications/Xcode-beta.app/Contents/Developer/usr/bin/make']
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.l3xdLR
bear: DEBUG: _split_command: input was: [u'g++', u'-c', u'-o', u'test.o', u'test.cpp']
bear: DEBUG: _split_command: output is: CompilationCommand(compiler='c++', phase=[u'-c'], flags=[], files=[u'test.cpp'], output=[u'test.o'])
bear: DEBUG: parse_exec_trace: parse exec trace file: /var/folders/jk/791tgt39495cz8kqz5kml7p00000gn/T/intercept-2vNj4w/execution.Uix5FZ
bear: DEBUG: _split_command: input was: [u'/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang', u'-cc1', u'-triple', u'x86_64-apple-macosx10.14.0', u'-Wdeprecated-objc-isa-usage', u'-Werror=deprecated-objc-isa-usage', u'-emit-obj', u'-mrelax-all', u'-disable-free', u'-disable-llvm-verifier', u'-discard-value-names', u'-main-file-name', u'test.cpp', u'-mrelocation-model', u'pic', u'-pic-level', u'2', u'-mthread-model', u'posix', u'-mdisable-fp-elim', u'-fno-strict-return', u'-masm-verbose', u'-munwind-tables', u'-target-cpu', u'penryn', u'-dwarf-column-info', u'-debugger-tuning=lldb', u'-target-linker-version', u'408.1.1', u'-coverage-notes-file', u'/Users/till/Desktop/test.gcno', u'-resource-dir', u'/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/10.0.0', u'-stdlib=libc++', u'-fdeprecated-macro', u'-fdebug-compilation-dir', u'/Users/till/Desktop', u'-ferror-limit', u'19', u'-fmessage-length', u'204', u'-stack-protector', u'1', u'-fstack-check', u'-fblocks', u'-fencode-extended-block-signature', u'-fobjc-runtime=macosx-10.14.0', u'-fcxx-exceptions', u'-fexceptions', u'-fmax-type-align=16', u'-fdiagnostics-show-option', u'-fcolor-diagnostics', u'-o', u'test.o', u'-x', u'c++', u'test.cpp']
```

That pre-release statement was targeted towards me using macOS 10.14 (beta) and Xcode 10 (beta) - both super early versions.

The symptoms indeed "feel" like I was using a different `g++` (which is apple's `clang++` frontend) with `bear`. But how can I determine that....

```
$ which g++
/usr/bin/g++
```
I have modified my `Makefile` to make sure  the compiler in use is determined by its path, not just its name;

```
test: test.o
	/usr/bin/g++ -o test test.o

test.o: test.cpp
	/usr/bin/g++ --version
	/usr/bin/g++ -c -o test.o test.cpp
```

Results are rather interesting - but also confusing to me right now;

```
$ bear make
/usr/bin/g++ --version
Configured with: --prefix=/Applications/Xcode-beta.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.10.25.5)
Target: x86_64-apple-darwin18.0.0
Thread model: posix
InstalledDir: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
/usr/bin/g++ -c -o test.o test.cpp
In file included from test.cpp:1:
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdio.h:108:15: fatal error: 'stdio.h' file not found
#include_next <stdio.h>
              ^~~~~~~~~
1 error generated.
make: *** [test.o] Error 1
```

```
$ make
/usr/bin/g++ --version
Configured with: --prefix=/Applications/Xcode-beta.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.10.25.5)
Target: x86_64-apple-darwin18.0.0
Thread model: posix
InstalledDir: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
/usr/bin/g++ -c -o test.o test.cpp
/usr/bin/g++ -o test test.o
```

Note how it is the very same compiler but its (internal) arguments suddenly change dramatically -- specifically `with-gxx-include-dir` suddenly changes towards something that won't fly on this system. So that is the problem, but why does it happen and how can I prevent that?

--- a/libear/ear.c
+++ b/libear/ear.c
@@ -54,7 +54,7 @@
 
 #if defined HAVE_NSGETENVIRON
 # include <crt_externs.h>
-static char **environ;
+#define environ (*_NSGetEnviron())
 #else
 extern char **environ;
 #endif
@@ -183,11 +183,6 @@ static void on_unload(void) {
 }
 
 static int mt_safe_on_load(void) {
-#ifdef HAVE_NSGETENVIRON
-    environ = *_NSGetEnviron();
-    if (0 == environ)
-        return 0;
-#endif
     // Capture current relevant environment variables
     return capture_env_t(&initial_env);
 }
@@ -531,7 +526,7 @@ static int write_report(int fd, char const *const argv[]) {
     return 0;
 }
 
-/* update environment assure that chilren processes will copy the desired
+/* updating the environment assures that child processes will copy the desired
  * behaviour */
 
 static int capture_env_t(bear_env_t *env) {

